onetl/file/format/xml.py

```diff
@@ -0,0 +1,228 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, ClassVar
+
+from pydantic import Field
+
+from onetl._util.java import try_import_java_class
+from onetl._util.scala import get_default_scala_version
+from onetl._util.spark import get_spark_version
+from onetl._util.version import Version
+from onetl.exception import MISSING_JVM_CLASS_MSG
+from onetl.file.format.file_format import ReadWriteFileFormat
+from onetl.hooks import slot, support_hooks
+
+if TYPE_CHECKING:
+    from pyspark.sql import SparkSession
+
+
+PROHIBITED_OPTIONS = frozenset(
+    (
+        # filled by onETL classes
+        "path",
+    ),
+)
+
+
+READ_OPTIONS = frozenset(
+    (
+        "rowTag",
+        "samplingRatio",
+        "excludeAttribute",
+        "treatEmptyValuesAsNulls",
+        "mode",
+        "inferSchema",
+        "columnNameOfCorruptRecord",
+        "attributePrefix",
+        "valueTag",
+        "charset",
+        "ignoreSurroundingSpaces",
+        "wildcardColName",
+        "rowValidationXSDPath",
+        "ignoreNamespace",
+        "timestampFormat",
+        "dateFormat",
+    ),
+)
+
+WRITE_OPTIONS = frozenset(
+    (
+        "rowTag",
+        "rootTag",
+        "declaration",
+        "arrayElementName",
+        "nullValue",
+        "attributePrefix",
+        "valueTag",
+        "compression",
+        "timestampFormat",
+        "dateFormat",
+    ),
+)
+
+
+log = logging.getLogger(__name__)
+
+
+@support_hooks
+class XML(ReadWriteFileFormat):
+    """
+    XML file format. |support_hooks|
+
+    Based on `Databricks Spark XML <https://github.com/databricks/spark-xml>`_ file format.
+
+    Supports reading/writing files with ``.xml`` extension.
+
+    .. versionadded:: 0.9.5
+
+    .. dropdown:: Version compatibility
+
+        * Spark versions: 3.2.x - 3.4.x.
+
+        * Scala versions: 2.12 - 2.13
+
+        * Java versions: 8 - 20
+
+
+        See documentation from link above.
+
+    .. note ::
+
+        You can pass any option to the constructor, even if it is not mentioned in this documentation.
+        **Option names should be in** ``camelCase``!
+
+        The set of supported options depends on Spark version. See link above.
+
+        **Important**: When **reading** files with timestamps, it may be necessary to specify the ``timestampFormat`` option to ensure dates are parsed correctly. Without it, date parsing may return ``null`` values. Example: ``timestampFormat="yyyy-MM-dd HH:mm:ssXXX"``.
```

Comment:
```suggestion
    .. warning::

        When **reading** files with timestamps, it may be necessary to specify the ``timestampFormat`` option to ensure dates are parsed correctly. Without it, date parsing may return ``null`` values. Example: ``timestampFormat="yyyy-MM-dd HH:mm:ssXXX"``.
```